aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -863,6 +893,134 @@ std::tuple<Tensor&,Tensor&> qr_out(Tensor& Q, Tensor& R, const Tensor& self, boo
   return std::tuple<Tensor&, Tensor&>(Q, R);
 }
 
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ syevd ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+template <typename scalar_t>
+static void apply_syevd(Tensor& w, Tensor& v, bool compute_v, std::string uplo_str, std::vector<int64_t>& infos) {
+#ifndef USE_LAPACK
+  AT_ERROR("syevd: LAPACK library not found in compilation");
+#else
+  using value_t = typename c10::scalar_value_type<scalar_t>::type;
+
+  auto v_data = v.data_ptr<scalar_t>();
+  auto w_data = w.data_ptr<value_t>();
+  auto v_matrix_stride = matrixStride(v);
+  auto w_stride = w.size(-1);
+  auto batch_size = batchCount(v);
+  auto n = v.size(-1);
+  auto lda = std::max(int64_t{1}, n);
+
+  char uplo = uplo_str == "U" ? 'U' : 'L';
+  char jobz = compute_v ? 'V' : 'N';
+
+  int info;
```

Comment:
Add a comment here explaining that the use of imprecise types, like int, is consistent with the LAPACK interface (otherwise we'd use a precise type, like int32_t).

Reply:
Nit: if we were being extremely correct we would add a using declaration to make it easy to change this int type later

Reply:
I think this kind of type change should happen at once for all functions that call to LAPACK. There should be `lapack_int` that depends on the type of the linked LAPACK or only 64-bit indexing should be used by default and only 64-bit version of LAPACK allowed.

Reply:
I agree with you and the comment looks great.